src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3992,7 +3996,7 @@ jint Arguments::parse(const JavaVMInitArgs* initial_cmd_args) {
       "DumpLoadedClassList is not supported in this VM\n");
     return JNI_ERR;
   }
-  if ((UseSharedSpaces && FLAG_IS_CMDLINE(UseSharedSpaces)) ||
+  if ((UseSharedSpaces && xshare_auto_cmd_line) ||
```

Comment:
xshare_auto_cmd_line is set with UseSharedSpaces (line 2715) so it should be OK to check only for itself.

Reply:
Hi Yumin, thanks for looking at this.  If someone had a command line that contained the equivalent of -Xshare:auto ... -Xshare:off then, if only xshare_auto_cmd_line was checked, they would get the warning even though UseSharedSpaces was false.  So, I'd like to keep the existing check.  Thanks, Harold